xbmc/platform/linux/WebOSTVPlatformConfig.cpp

```diff
@@ -75,3 +81,59 @@ bool WebOSTVPlatformConfig::SupportsHDR()
 {
   return ms_config[SUPPORT_HDR].asBoolean();
 }
+
+void WebOSTVPlatformConfig::LoadARCStatus()
+{
+  HContext requestContext;
+  requestContext.pub = true;
+  requestContext.multiple = false;
+  requestContext.callback = [](LSHandle* sh, LSMessage* msg, void* ctx) -> bool
+  {
+    std::string message = HLunaServiceMessage(msg);
+    CLog::LogF(LOGDEBUG, "ARC controller: {}", message);
+
+    CVariant parsed;
+    if (!CJSONVariantParser::Parse(message, parsed))
+    {
+      CLog::LogF(LOGERROR, "Failed to parse ARC controller JSON");
+      return false;
+    }
+
+    const std::string arcActivation = parsed[EARC_ACTIVATION].asString();
+    bool ddplusSupported = false;
+
+    if (arcActivation == EARC)
+    {
+      const CVariant& caps = parsed[EARC_CAPABILITY];
+      if (caps.isArray())
+      {
+        for (unsigned int i = 0; i < caps.size(); ++i)
+        {
+          if (caps[i].asString() == DDPLUS)
+          {
+            ddplusSupported = true;
+            break;
+          }
+        }
+      }
+    }
+
+    ms_config[SUPPORT_EARCDDPlus] = ddplusSupported;
+
+    return true;
+  };
+
+  std::string payload = "{}";
+  constexpr const char* LUNA_ARC_STATUS = "luna://com.webos.service.arccontroller/getARCState";
+
+  if (HLunaServiceCall(LUNA_ARC_STATUS, payload.c_str(), &requestContext))
+  {
+    CLog::LogF(LOGWARNING, "Luna ARC controller request call failed");
+  }
+}
+
+bool WebOSTVPlatformConfig::SupportsEARCDDPlus()
+{
+  LoadARCStatus();
```

Comment:
Don't query here always. Also the luna call is not guaranteed to finish synchronously so this can be potentially unsafe